### PR TITLE
Add ClusterEnvironment healthcheck

### PR DESCRIPTION
**Pull request by baijum**
Ref. https://issues.redhat.com/browse/APPSVC-1282

### CODE REVIEW

File: controllers/clusterenvironment_controller.go [hunk 1]
```diff
@@ -176,6 +178,49 @@ func (r *ClusterEnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return ctrl.Result{}, nil
 }
 
+func (r *ClusterEnvironmentReconciler) MonitorHealth() {
+	ctx := context.Background()
+	l := ctrl.Log.WithName("healthcheck")
+	for {
+		ceList := &primazaiov1alpha1.ClusterEnvironmentList{}
+		if err := r.List(ctx, ceList, &client.ListOptions{Namespace: constants.PrimazaNamespace}); err != nil {
```
Comment (filariow) [inline controllers/clusterenvironment_controller.go:186]: Replace with WATCH_NAMESPACE value, https://github.com/primaza/primaza/blob/c381d7a89f5d9360fb9b406ed4a46ed7d57bb718/config/manager/manager.yaml#L103

Comment (baijum) [reply controllers/clusterenvironment_controller.go]: Fixed.

File: controllers/clusterenvironment_controller.go [hunk 2]
```diff
@@ -176,6 +177,49 @@ func (r *ClusterEnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return ctrl.Result{}, nil
 }
 
+func (r *ClusterEnvironmentReconciler) MonitorHealth(ctx context.Context, ns string, hci int) {
```
Comment (filariow) [inline controllers/clusterenvironment_controller.go:180]: I'd propose to move this in `pkg/primaza/healthchecks/clusterenvironment_healthchecker.go`

File: cmd/primaza/main.go [hunk 3]
```diff
@@ -162,17 +166,44 @@ func main() {
...
+	cer *controllers.ClusterEnvironmentReconciler
+	ns  string
+	hci int
+}
+
+func (h *clusterEnvironmentHealthcheck) Start(ctx context.Context) error {
+	go h.cer.MonitorHealth(ctx, h.ns, h.hci)
+	return nil
+}
+
```
Comment (filariow) [inline cmd/primaza/main.go:201]: I'd propose to move this in `pkg/primaza/healthchecks/clusterenvironment_healthchecker.go`

**Review by filariow — CHANGES_REQUESTED** [review]
some feedback

File: cmd/primaza/main.go [hunk 4]
```diff
@@ -162,17 +166,44 @@ func main() {
 		os.Exit(1)
 	}
 
+	var hci int
+	if hci, err = strconv.Atoi(cfg.HealthCheckInterval); err != nil {
+		hci = 600
+		setupLog.Error(err, "default health check interval not set")
+	}
+
```
[the remark this answers is not in the document]
Comment (baijum) [reply cmd/primaza/main.go]: I used `MinimumHealtCheckInterval` as the return value in the second condition. Let me know, if that's okay.

File: cmd/primaza/main.go [hunk 5]
```diff
@@ -191,10 +235,13 @@ func getConfig() (*config, error) {
 		return nil, err
 	}
 
+	hci, _ := getRequiredEnv(EnvHealthCheckInterval)
```
Comment (filariow) [inline cmd/primaza/main.go:238]: I'd propose the following change to have the value in config set to the actual value we are using for Healthcheck interval. WDYT?

```suggestion
	hci := getHealthCheckIntervalFromEnv(log)
```

and having

```
func getHealtcheckIntervalFromEnv(log logr.Logger) int {
        ehci := os.Getenv(EnvHealthCheckInterval)
	hci, err := strconv.Atoi(ehci)
	if err != nil {
		log.Info("HealthCheckInterval not set or not a integer value: using default value", "interval", cfg.HealthCheckInterval, "error", err.Error(), "default", DefaultHealthCheckInterval)
		return DefaultHealthCheckInterval
	}
	if hci <= MinimumHealtCheckInterval {
		log.Info("provided HealthCheckInterval lower than minimum: using minimum value", "interval", cfg.HealthCheckInterval, "minimum", MinimumHealtCheckInterval, "default", DefaultHealthCheckInterval)
		return MinimumHealtCheckInterval
	}

	return hci
}
```

File: test/acceptance/features/steps/primazacluster.py [hunk 6]
```diff
@@ -133,3 +134,9 @@ def ensure_application_namespace_exists(
         context, cluster_name: str, namespace: str, cluster_environment: str, tenant: str = "primaza-system"):
     primaza = context.cluster_provider.get_primaza_cluster(cluster_name)  # type: PrimazaCluster
     primaza.create_application_namespace(namespace, tenant, cluster_environment)
+
+
+@step(u'On Primaza Cluster "{cluster_name}", sleep for {time_interval:d} minutes')
+def ensure_to_sleep_for_certain_time_interval(context, cluster_name: str, time_interval: int):
+    context.cluster_provider.get_primaza_cluster(cluster_name)
+    time.sleep(time_interval*60)
```
Comment (filariow) [inline test/acceptance/features/steps/primazacluster.py:142]: we already have one step like this at https://github.com/primaza/primaza/blob/ec14c4d84e0ae53c3e3fbf683da003f2f4abde62/test/acceptance/features/steps/util.py#L31

**Review by filariow — CHANGES_REQUESTED** [review]
a couple of nits. other than that LGTM

File: cmd/primaza/main.go (hunk 5 above)
Comment (baijum) [reply cmd/primaza/main.go]: Done

File: test/acceptance/features/steps/primazacluster.py (hunk 6 above)
Comment (baijum) [reply test/acceptance/features/steps/primazacluster.py:142]: Done

**Review by filariow — APPROVED** [review]
lgtm

**Pull request merged**